### PR TITLE
CSI-1027 near integration

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

**Comment by irshadCypherock** [comment]
Task linked: [CSI-1027 Near Coin Intergration in CySync](https://app.clickup.com/t/37308523/CSI-1027)

File: src/flows/receiveTransaction/index.ts [hunk 1]
```diff
@@ -39,6 +43,7 @@ enum RECEIVE_TRANSACTION_STATUS {
   RECV_TXN_DERIVE_ADD_SCREEN,
   RECV_TXN_DERIVE_ADD,
   RECV_TXN_DISPLAY_ADDR,
+  RECV_TXN_DISPLAY_ADDR_NEAR,
```
Comment (irshadCypherock) [inline src/flows/receiveTransaction/index.ts]: Why is the BTC enum changed?

File: src/flows/receiveTransaction/index.ts [hunk 2]
```diff
@@ -102,8 +132,15 @@ export class TransactionReceiver extends CyFlow {
...
 
+    let nearBlockVerify = false;
     if (data.commandType === 65 && data.data === '01') {
       this.emit('coinsConfirmed', true);
+      if (coin.group === CoinGroup.Near && customAccount) {
+        nearBlockVerify = true;
+      }
+    } else if (data.commandType === 65 && data.data === '02') {
+      this.emit('coinsConfirmed', true);
+      this.emit('customAccountExists', true);
```
Comment (irshadCypherock) [inline src/flows/receiveTransaction/index.ts]: I don't think we need to implement NEAR in legacy command types.

File: src/flows/receiveTransaction/index.ts [hunk 3]
```diff
@@ -230,11 +304,209 @@ export class TransactionReceiver extends CyFlow {
...
+          }
+
+          if (
+            pinExists &&
+            status.flowStatus >= pinEnteredCmdStatus &&
+            pinEnteredState === 0
+          ) {
+            pinEnteredState = 1;
+          }
+
```
Comment (irshadCypherock) [inline src/flows/receiveTransaction/index.ts]: When there are multiple waitForCommandOutput function, it's better to implement onStatus function once and use it in all waitForCommandOutput calls. Check sendTransaction.

File: src/flows/sendTransaction/index.ts [hunk 4]
```diff
@@ -259,6 +284,32 @@ export class TransactionSender extends CyFlow {
...
+      } catch (error) {
+        this.emit('signatureVerify', {
+          isVerified: false,
+          index: -1,
+          error
+        });
+      }
+
+      logger.info('Signed txn', { signedTxnNear });
+      this.emit('signedTxn', signedTxnNear);
```
Comment (irshadCypherock) [inline src/flows/sendTransaction/index.ts]: No need to implement this in Legacy commands

File: src/flows/sendTransaction/index.ts [hunk 5]
```diff
@@ -823,7 +967,7 @@ export class TransactionSender extends CyFlow {
           const res = await Server.eth.transaction
             .getFees({ network })
             .request();
-          feeRate = Math.round(res.data / 1000000000);
+          feeRate = res.data.FastGasPrice;
```
Comment (irshadCypherock) [inline src/flows/sendTransaction/index.ts]: Any reason for this change?

**Review by irshadCypherock — CHANGES_REQUESTED** [review]

File: src/flows/receiveTransaction/index.ts (hunk 3 above)
Comment (ghost) [reply src/flows/receiveTransaction/index.ts]: onStatus function for near is not the same as others it contains segmented part of original onStatus function but not the same

File: src/flows/sendTransaction/index.ts [hunk 6]
```diff
@@ -352,7 +381,18 @@ export class TransactionSender extends CyFlow {
         SEND_TRANSACTION_STATUS_ETH.SEND_TXN_CONFIRM_PASSPHRASE_ETH;
       pinEnteredCmdStatus = SEND_TRANSACTION_STATUS_ETH.SEND_TXN_ENTER_PIN_ETH;
       cardTapCmdStatus =
-        SEND_TRANSACTION_STATUS_ETH.SEND_TXN_TAP_CARD_SEND_CMD_ETH;
+        SEND_TRANSACTION_STATUS_NEAR.SEND_TXN_TAP_CARD_SEND_CMD_NEAR;
```
Comment (irshadCypherock) [inline src/flows/sendTransaction/index.ts]: This looks incorrect

**Review by irshadCypherock — CHANGES_REQUESTED** [review]

File: src/flows/sendTransaction/index.ts (hunk 5 above)
Comment (irshadCypherock) [reply src/flows/sendTransaction/index.ts]: @TejasvCypherock ?

File: src/flows/receiveTransaction/index.ts (hunk 3 above)
Comment (irshadCypherock) [reply src/flows/receiveTransaction/index.ts]: So we can create an onStatus function for near and use it in the two functions, right?

File: src/flows/sendTransaction/index.ts (hunk 5 above)
Comment (ghost) [reply src/flows/sendTransaction/index.ts]: I reverted this it seems I accidentally changed it during a merge

**Review by irshadCypherock — DISMISSED** [review]

**Pull request merged**